File: Datasets/TrickyBugs/GenProgs/tc_generated_progs_cpp/p03014/p03014_num2_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int h, w;
    cin >> h >> w;
    vector<string> s(h);
    for(int i=0; i<h; i++) {
        cin >> s[i];
    }
    vector<vector<int>> cnt(h, vector<int>(w));
    
    // Count the number of unobstructed squares to the left of each square
    for(int i=0; i<h; i++) {
        int l = 0; // number of unobstructed squares so far
        for(int j=0; j<w; j++) {
            if(s[i][j] == '#') {
                for(int k=j-l; k<j; k++) {
                    cnt[i][k] += l;
                }
                l = 0;
            } else {
                l++;
            }
        }
    }
    
    // Count the number of unobstructed squares above each square
    for(int j=0; j<w; j++) {
        int l = 0; // number of unobstructed squares so far
        for(int i=0; i<h; i++) {
            if(s[i][j] == '#') {
                for(int k=i-l; k<i; k++) {
                    cnt[k][j] += l;
                }
                l = 0;
            } else {
                l++;
            }
        }
    }
    
    int answer = 0;
    // Find the maximum number of squares that can be lit up by a lamp
    for(int i=0; i<h; i++) {
        for(int j=0; j<w; j++) {
            int range = cnt[i][j] - 1;
            answer = max(answer, range);
        }
    }
    
    cout << answer << endl;
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03014/p03014_num2_parsed.py
H, W = map(int, input().split())
S = [input() for _ in range(H)]

max_lighted_squares = 0

# For each cell in the grid, find the maximum number of lighted squares
for i in range(H):
    for j in range(W):
        if S[i][j] == '.':
            # Initialize counts for each direction
            count_up = 1
            count_down = 1
            count_left = 1
            count_right = 1
            
            # Check the up direction
            r = i - 1
            while r >= 0 and S[r][j] != '#':
                count_up += 1
                r -= 1

            # Check the down direction
            r = i + 1
            while r < H and S[r][j] != '#':
                count_down += 1
                r += 1

            # Check the left direction
            c = j - 1
            while c >= 0 and S[i][c] != '#':
                count_left += 1
                c -= 1

            # Check the right direction
            c = j + 1
            while c < W and S[i][c] != '#':
                count_right += 1
                c += 1

            # Calculate the total number of lighted squares for this cell
            total = count_up + count_down + count_left + count_right - 3

            # Update the maximum number of lighted squares
            max_lighted_squares = max(max_lighted_squares, total)

# Print the maximum possible number of squares lighted by the lamp
print(max_lighted_squares)
